util/src/main/java/io/camunda/zeebe/util/sched/ActorScheduler.java

```diff
@@ -52,19 +50,15 @@ public ActorFuture<Void> submitActor(final Actor actor) {
    * @param schedulingHints additional scheduling hint
    */
   @Override
-  public ActorFuture<Void> submitActor(final Actor actor, final int schedulingHints) {
+  public ActorFuture<Void> submitActor(final Actor actor, final SchedulingHints schedulingHints) {
     checkRunningState();
 
     final ActorTask task = actor.actor.task;
 
-    final ActorFuture<Void> startingFuture;
-    if (SchedulingHints.isCpuBound(schedulingHints)) {
-      task.setPriority(SchedulingHints.getPriority(schedulingHints));
-      startingFuture = actorTaskExecutor.submitCpuBound(task);
-    } else {
-      startingFuture = actorTaskExecutor.submitIoBoundTask(task);
-    }
-    return startingFuture;
+    return switch (schedulingHints) {
+      case CPU_BOUND -> actorTaskExecutor.submitCpuBound(task);
+      case IO_BOUND -> actorTaskExecutor.submitIoBoundTask(task);
```

Comment:
:question: what happens on default here? Or will this break the compiling if new type is added?

We could also just have two methods `submitIOBoundActor` and `submitActor` :shrug:

Reply:
Yes, that'd be an error. It's exhaustive matching :heart_eyes: 

> We could also just have two methods submitIOBoundActor and submitActor

True! I like this api so unless you feel strongly about it I'd leave it as is.

Reply:
Great I love compile errors :)